### PR TITLE
Update upstream mbedTLS patches backported to v2.16

### LINKED ISSUE #3170: mbedtls_x509_write_csr fails with quote extension
Seems @CodeMonkeyLeet  worked around this for writing certificates with this patch https://github.com/openenclave/openenclave/blob/040beb10dba521ee95af10b0898707e61cceb3b6/3rdparty/mbedtls/0001-Patch-x509write_crt.c-for-attestedTLS.patch

We need the same for CSRs :)

### PR DESCRIPTION
**Pull request by CodeMonkeyLeet**
- Add 0001-Avoid-stack-allocation-of-large-memory-buffers.patch from upstream [mbedTLS PR #3464](https://github.com/ARMmbed/mbedtls/pull/3464).
- Replace 0001-Patch-x509_write_crt.c-for-attestedTLS.patch with patch from upstream mbedTLS 0001-Backport-2632-code-changes-to-2.16.patch. This contains commits 6ad3fd1, 4063ad2 and 67d4259 from [mbedTLS PR #2632](https://github.com/ARMmbed/mbedtls/pull/2632).
- Apply the upstream changes to 3rdparty/mbedtls.

Fixes #3170

Signed-off-by: Simon Leet <simon.leet@microsoft.com>

### CODE REVIEW

**Comment by anakrish** [comment]
The changes look good at first glance.
Some discussion about the process:
- Should we continue to maintain these patches or just switch to using forks with patches applied?
- The patch to upstream ([mbedTLS PR #3464](https://github.com/ARMmbed/mbedtls/pull/3464)) needs to be from a fork under `openenclave` organization.

**Comment by CodeMonkeyLeet** [comment]
@anakrish I ran the option by @radhikaj before doing this, but the prioritization is to get the fix for #3170 in and not change the build process for mbedTLS for v0.10.

File: 3rdparty/mbedtls/0001-Backport-2632-code-changes-to-2.16.patch [hunk 1]
```diff
@@ -0,0 +1,505 @@
...
++                          mbedtls_pk_write_pubkey( &c, buf, ctx->issuer_key ) );
+
+     ret = mbedtls_sha1_ret( buf + sizeof( buf ) - len, len,
+                             buf + sizeof( buf ) - 20 );
+@@ -243,15 +256,19 @@ int mbedtls_x509write_crt_set_authority_key_identifier( mbedtls_x509write_cert *
+     len = 20;
+
+     MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_len( &c, buf, len ) );
+-    MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_tag( &c, buf, MBEDTLS_ASN1_CONTEXT_SPECIFIC | 0 ) );
++    MBEDTLS_ASN1_CHK_ADD( len,
```
Comment (anakrish) [inline 3rdparty/mbedtls/0001-Backport-2632-code-changes-to-2.16.patch:193]: Some of the diffs seem to be just code formatting. Can you please double-check if we are running clang-format incorrectly on 3rparty code.

Comment (CodeMonkeyLeet) [reply 3rdparty/mbedtls/0001-Backport-2632-code-changes-to-2.16.patch:193]: These are not clang-formatted to OE SDK project, these are the result of commit 6ad3fd1 in the upstream mbedtls project.

File: 3rdparty/mbedtls/0001-Backport-2632-code-changes-to-2.16.patch [hunk 2]
```diff
@@ -0,0 +1,505 @@
...
+     if( ctx->version == MBEDTLS_X509_CRT_VERSION_3 )
+     {
+-        MBEDTLS_ASN1_CHK_ADD( len, mbedtls_x509_write_extensions( &c, tmp_buf, ctx->extensions ) );
+-        MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_len( &c, tmp_buf, len ) );
+-        MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_tag( &c, tmp_buf, MBEDTLS_ASN1_CONSTRUCTED |
+-                                                           MBEDTLS_ASN1_SEQUENCE ) );
+-        MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_len( &c, tmp_buf, len ) );
+-        MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_tag( &c, tmp_buf, MBEDTLS_ASN1_CONTEXT_SPECIFIC |
+-                                                           MBEDTLS_ASN1_CONSTRUCTED | 3 ) );
++        MBEDTLS_ASN1_CHK_ADD( len,
```
Comment (anakrish) [inline 3rdparty/mbedtls/0001-Backport-2632-code-changes-to-2.16.patch:298]: In this section for example, there does not seem to be any code difference; only format changes.

Comment (CodeMonkeyLeet) [reply 3rdparty/mbedtls/0001-Backport-2632-code-changes-to-2.16.patch:298]: Same observation as above, there are explicit formatting changes as part of the upstream PR. If you look closely, you'll see they don't match our clang-format rules.

**Pull request merged**